src/Shared/UnitTests/FileMatcher_Tests.cs

```diff
@@ -1241,6 +1241,7 @@ public void IllegalPaths()
             ValidateIllegal("http://www.website.com");
         }
 
+#if !FEATURE_MSIOREDIST
```

Comment:
This is the unit test that I broke. I must say, I am not quite sure about it. Behavior of GetFiles changed in this test. Before the change it returned the input verbatim and now it returns an empty array. Which one is a correct behavior is a question.

The reason of the change is that Directory.EnumerateFiles no more throws exception when the filespec pattern is too long, it allows much longer strings.
As in the summary comment to the function `GetFiles` it is stated "never throw IO exceptions: if there is no match, returns the input verbatim."  If it is just no matches, it returns just empty array. I have not found more specific approach to define when to return the "input verbatim" and when to return an empty array.

Reply:
How is this passing, exactly? Shouldn't it be failing when it actually hits the OS APIs? Or is it passing in some cases because the OS flag to allow long paths is turned on?

Reply:
Hmm, it runs only for Windows platform. So, it does not calls at all in the result. I suppose I can just delete this test. The question about this test, however, stays: does the function still behaves well.

Reply:
I am not sure which errors could be hit when it it fails with exception, but no-exceptions behavior is asked in the func description.

Reply:
> How is this passing
Well, for .NET Framework GetFiles catches IO exception about the long path and returns filespec as the only match. And I am not quite sure that was correct behavior, as it is written in the test that it is only to avoid a crash. Unfortunately, I cannot go to the history of the test there and find more info about a crash: it is initial commit

Reply:
You're looking for (internal link, VPN required): https://vstfdevdiv/DevDiv2/DevDiv/_versionControl/changeset/65580/